components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/internal/SecretManagerComponent.java

```diff
@@ -0,0 +1,139 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.secret.mgt.core.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.secret.mgt.core.SecretManager;
+import org.wso2.carbon.identity.secret.mgt.core.SecretManagerImpl;
+import org.wso2.carbon.identity.secret.mgt.core.SecretResolveManager;
+import org.wso2.carbon.identity.secret.mgt.core.SecretResolveManagerImpl;
+import org.wso2.carbon.identity.secret.mgt.core.dao.SecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.dao.impl.CachedBackedSecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.dao.impl.SecretDAOImpl;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.DB_TABLE_NAME;
+
+/**
+ * OSGi declarative services component which handles registration and un-registration of configuration management
+ * service.
+ */
+@Component(
+        name = "carbon.secret.mgt.component",
+        immediate = true
+)
+public class SecretManagerComponent {
+
+    private static final Log log = LogFactory.getLog(SecretManagerComponent.class);
+    private List<SecretDAO> secretDAOS = new ArrayList<>();
+
+    /**
+     * Register SecretManager as an OSGI service.
+     *
+     * @param componentContext OSGI service component context.
+     */
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+
+        try {
+            BundleContext bundleContext = componentContext.getBundleContext();
+
+            SecretDAO secretDAO = new SecretDAOImpl();
+            SecretManagerComponentDataHolder.getInstance().setSecretDAOS(secretDAOS);
+            bundleContext.registerService(SecretDAO.class.getName(),
+                    new CachedBackedSecretDAO(secretDAO), null);
+            bundleContext.registerService(SecretManager.class.getName(),
+                    new SecretManagerImpl(), null);
+            bundleContext.registerService(SecretResolveManager.class.getName(),
+                    new SecretResolveManagerImpl(), null);
+            SecretManagerComponentDataHolder.getInstance().setSecretManagementEnabled
+                    (isSecretManagementEnabled());
+        } catch (Throwable e) {
+            log.error("Error while activating SecretManagerComponent.", e);
```

Comment:
Better to through the exception out. So that the server startup will be blocked if this component activation fails

Reply:
we can remove the try-catch block inside the activate methods as it catch the Throwable

Reply:
resolved